src/_pytest/config/__init__.py

```diff
@@ -65,10 +65,17 @@
 from _pytest.pathlib import resolve_package_path
 from _pytest.pathlib import safe_exists
 from _pytest.stash import Stash
+from _pytest.stash import StashKey
 from _pytest.warning_types import PytestConfigWarning
 from _pytest.warning_types import warn_explicit_for
 
 
+# File descriptor for stdout, duplicated before capture starts.
+# This allows the terminal reporter to bypass pytest's output capture (#8973).
+# The FD is duplicated early in _prepareconfig before any capture can start.
+stdout_fd_dup_key = StashKey[int]()
```

Comment:
It seems weird to me that this is in config/ when config doesn't really care about it. It should be in terminal.py or maybe capture.py if we want to take care of it "at the source".